### DISCUSSION TITLE
Music Composition with Deep Learning (Project 243)

### DISCUSSION THREAD

**Discussion by robertogl** [mathworks/MATLAB-Simulink-Challenge-Project-Hub#78] [Collaborate]
Contribute to the discussion by asking and/or answering questions, commenting, or sharing your ideas for solutions to [project 243](https://github.com/mathworks/MathWorks-Excellence-in-Innovation/blob/main/projects/Music%20Composition%20with%20Deep%20Learning)

---
*:robot: An AI bot may reply to your comments with suggestions and guidance. It will not solve the project for you. Your comments may be processed by a third-party LLM. To direct a question to a specific person, tag them with @username.*

**Comment by luciry** [comment]
Hello, my name is Luca, I study Strategic Engineering in the University of Genova. I'm interested in the project and I will work on it for my exam of Machine Learning and Data Analysis

**Reply by cps13** [reply]
Hi Luca, good luck with the project. I'm excited to see how it progresses!

**Reply by luciry** [reply]
Thank you very much! 😄

**Comment by Akshada07pute** [comment]
Can we do this project along with some hardware like microcontroller and speaker

**Reply by cps13** [reply]
That sounds like a fun idea! You are welcome to expand the project description however you would like to.

**Reply by Akshada07pute** [reply]
As we get the output in the musical sound how can we make the output coming from the speaker which is interfaced with microcontroller  can you please suggest some idea

**Reply by cps13** [reply]
This will depend on what exactly the output of your model is. In the project description, we recommend outputting a MIDI sequence that can then be played in a 3rd party tool, or you can synthesize audio with the Audio Toolbox. That could be a good place to start:

- Play the resultant sequence, either by writing a MIDI file and playing it outside of MATLAB, or synthesizing audio with the [Audio Toolbox™](https://www.mathworks.com/help/audio/musical-instrument-digital-interface-midi.html?s_tid=CRUX_lftnav).

**Comment by Arnav2Prasad** [comment]
Hi, I'm Arnav Prasad, a BTech CSE student at COEP Technological University, Pune, India.

Is the problem statement : "Develop a solution where a user inputs an image of musical notation, and the system identifies the notes and plays the corresponding tune" aligns in to yours problem statement?

**Reply by cps13** [reply]
It is a little different, but that sounds like a great project, too!

**Reply by Arnav2Prasad** [reply]
Okay, I’ll definitely make some progress in this project and would love the opportunity to present it to you. Looking forward to our continued discussion!

**Comment by TanV-04** [comment]
Hi, I'm Tanvi! I'm a third-year CSE undergrad at SIT, Pune, India.

My teammates and I completed this project as part of our MATLAB elective in college, developing a basic yet efficient version based on the concepts and techniques we learned in class.

Here's the GitHub repository link: https://github.com/TanV-04/ESMT_Project/tree/master

We’d love to hear your feedback and suggestions to help us improve further!

**Reply by aksh-sush** [reply]
Hey! I'm really excited to check it out. However, when I cloned your repo, it didn’t show me any files. Did I maybe copy the wrong branch?

**Reply by TanV-04** [reply]
Try downloading the zip, if cloning doesn't work.

**Comment by sohamkorade** [comment]
From the project description:

> Use the [Deep Learning Toolbox™](https://www.mathworks.com/products/deep-learning.html) and the [Audio Toolbox™](https://www.mathworks.com/help/audio/musical-instrument-digital-interface-midi.html?s_tid=CRUX_lftnav) to create a generative model that can compose and play back a melody.

Is using the Deep Learning Toolbox™ or Audio Toolbox™ compulsory for the project? Can I use other libraries such as PyTorch for the deep learning model?

**Reply by robertogl** [reply]
using MATLAB is a requirement, unless you need features not available in MATLAB. In that case you can still use MATLAB in combination with other products.

**Comment by aksh-sush** [comment]
Hi, I’m a computer engineering student from India, and I’m really interested in visualizing music and much more. I’ve worked on analyzing and synthesizing Indian music, but since I’m new to MATLAB, everything still feels pretty raw and a bit intimidating. I’m still figuring out how to build a proper toolkit. So out of curiousity—what do you all use, and for what purposes? Are there any plugins or tools you’d consider absolute must-haves?

**Reply by TanV-04** [reply]
Some toolboxes you can use are:
- Signal Processing
- Audio Toolbox
- Image Processing Toolbox
- Spectrogram
- Musical Toolbox

These should help you get started.

**Reply by aksh-sush** [reply]
I couldn’t find the Spectrogram and Musical Toolbox tools. Anyway, instead of the Musical Toolbox, can I use the Signal Processing Toolbox? I’ll have to refine the audio and feed it into some ML, then work with it key by key.

**Reply by TanV-04** [reply]
Yeah, that works too. Its all about experimentation.

**Comment by aksh-sush** [comment]
hey hi! um i was browsing in wiki rewards to see standard rewards and under current project submission there's no "Music Composition with Deep Learning" does that simply mean that there havnt been any submissions there or has it been removed from the official list of topics? (just making sure)

**Reply by robertogl** [reply]
HI @aksh-sush,

There have been a few submissions with some issues that have not been posted. However, there are a few other promising ones under review.
